### ISSUE TITLE
make process fail when turning on : UA_ENABLE_WEBSOCKET_SERVER

### ISSUE THREAD

**Issue by jackybek** [open62541/open62541#4338]
## Description
ccmake option : 
UA_ENABLE_WEBSOCKET_SERVER= ON

## Background Information / Reproduction Steps
When making open62541, it fails:

pi@OPCServer-109:~/open62541-1.2.1/build $ make
[  4%] Built target open62541-generator-transport
[  8%] Built target open62541-generator-types
[ 13%] Built target open62541-generator-namespace
[ 30%] Built target open62541-amalgamation-header
[ 39%] Built target open62541-generator-statuscode
[ 60%] Built target open62541-amalgamation-source
[ 60%] Built target open62541-code-generation
[ 65%] Building C object CMakeFiles/open62541-object.dir/open62541.c.o
**/home/pi/open62541-1.2.1/build/open62541.c: In function ‘ServerNetworkLayerWS_start’:
/home/pi/open62541-1.2.1/build/open62541.c:91404:9: error: ‘struct lws_context_creation_info’ has no member named ‘ws_ping_pong_interval’
     info.ws_ping_pong_interval = 10;**
         ^
/home/pi/open62541-1.2.1/build/open62541.c: At top level:
cc1: error: unrecognized command line option ‘-Wno-static-in-inline’ [-Werror]
cc1: all warnings being treated as errors
make[2]: *** [CMakeFiles/open62541-object.dir/build.make:312: CMakeFiles/open62541-object.dir/open62541.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:305: CMakeFiles/open62541-object.dir/all] Error 2
make: *** [Makefile:152: all] Error 2

Used CMake options:

pi@OPCServer-109:~/open62541-1.2.1/build $ cmake -LA
CMake Warning:
  No source or binary directory provided.  Both will be assumed to be the
  same as the current working directory, but note that this warning will
  become a fatal error in future CMake releases.

CMake Error: The source directory "/home/pi/open62541-1.2.1/build" does not appear to contain CMakeLists.txt.
Specify --help for usage, or press the help button on the CMake GUI.
-- Cache values
BIBER_COMPILER:FILEPATH=BIBER_COMPILER-NOTFOUND
BIBTEX_COMPILER:FILEPATH=BIBTEX_COMPILER-NOTFOUND
BUILD_SHARED_LIBS:BOOL=OFF
CLANG_FORMAT_EXE:FILEPATH=CLANG_FORMAT_EXE-NOTFOUND
CLANG_FORMAT_PROGRAM:FILEPATH=CLANG_FORMAT_PROGRAM-NOTFOUND
CLANG_TIDY_PROGRAM:FILEPATH=CLANG_TIDY_PROGRAM-NOTFOUND
CMAKE_AR:FILEPATH=/usr/bin/ar
CMAKE_BUILD_TYPE:STRING=Debug
CMAKE_COLOR_MAKEFILE:BOOL=ON
CMAKE_CXX_COMPILER:FILEPATH=/usr/bin/c++
CMAKE_CXX_COMPILER_AR:FILEPATH=/usr/bin/gcc-ar-8
CMAKE_CXX_COMPILER_RANLIB:FILEPATH=/usr/bin/gcc-ranlib-8
CMAKE_CXX_FLAGS:STRING=
CMAKE_CXX_FLAGS_DEBUG:STRING=-g
CMAKE_CXX_FLAGS_MINSIZEREL:STRING=-Os -DNDEBUG
CMAKE_CXX_FLAGS_RELEASE:STRING=-O3 -DNDEBUG
CMAKE_CXX_FLAGS_RELWITHDEBINFO:STRING=-O2 -g -DNDEBUG
CMAKE_C_COMPILER:FILEPATH=/usr/bin/cc
CMAKE_C_COMPILER_AR:FILEPATH=/usr/bin/gcc-ar-8
CMAKE_C_COMPILER_RANLIB:FILEPATH=/usr/bin/gcc-ranlib-8
CMAKE_C_FLAGS:STRING=
CMAKE_C_FLAGS_DEBUG:STRING=-g
CMAKE_C_FLAGS_MINSIZEREL:STRING=-Os -DNDEBUG
CMAKE_C_FLAGS_RELEASE:STRING=-O3 -DNDEBUG
CMAKE_C_FLAGS_RELWITHDEBINFO:STRING=-O2 -g -DNDEBUG
CMAKE_EXE_LINKER_FLAGS:STRING=
CMAKE_EXE_LINKER_FLAGS_DEBUG:STRING=
CMAKE_EXE_LINKER_FLAGS_MINSIZEREL:STRING=
CMAKE_EXE_LINKER_FLAGS_RELEASE:STRING=
CMAKE_EXE_LINKER_FLAGS_RELWITHDEBINFO:STRING=
CMAKE_EXPORT_COMPILE_COMMANDS:BOOL=OFF
CMAKE_INSTALL_BINDIR:PATH=bin
CMAKE_INSTALL_DATADIR:PATH=
CMAKE_INSTALL_DATAROOTDIR:PATH=share
CMAKE_INSTALL_DOCDIR:PATH=
CMAKE_INSTALL_INCLUDEDIR:PATH=include
CMAKE_INSTALL_INFODIR:PATH=
CMAKE_INSTALL_LIBDIR:PATH=lib
CMAKE_INSTALL_LIBEXECDIR:PATH=libexec
CMAKE_INSTALL_LOCALEDIR:PATH=
CMAKE_INSTALL_LOCALSTATEDIR:PATH=var
CMAKE_INSTALL_MANDIR:PATH=
CMAKE_INSTALL_OLDINCLUDEDIR:PATH=/usr/include
CMAKE_INSTALL_PREFIX:PATH=/usr/local
CMAKE_INSTALL_RUNSTATEDIR:PATH=
CMAKE_INSTALL_SBINDIR:PATH=sbin
CMAKE_INSTALL_SHAREDSTATEDIR:PATH=com
CMAKE_INSTALL_SYSCONFDIR:PATH=etc
CMAKE_LINKER:FILEPATH=/usr/bin/ld
CMAKE_MAKE_PROGRAM:FILEPATH=/usr/bin/make
CMAKE_MODULE_LINKER_FLAGS:STRING=
CMAKE_MODULE_LINKER_FLAGS_DEBUG:STRING=
CMAKE_MODULE_LINKER_FLAGS_MINSIZEREL:STRING=
CMAKE_MODULE_LINKER_FLAGS_RELEASE:STRING=
CMAKE_MODULE_LINKER_FLAGS_RELWITHDEBINFO:STRING=
CMAKE_NM:FILEPATH=/usr/bin/nm
CMAKE_OBJCOPY:FILEPATH=/usr/bin/objcopy
CMAKE_OBJDUMP:FILEPATH=/usr/bin/objdump
CMAKE_RANLIB:FILEPATH=/usr/bin/ranlib
CMAKE_SHARED_LINKER_FLAGS:STRING=
CMAKE_SHARED_LINKER_FLAGS_DEBUG:STRING=
CMAKE_SHARED_LINKER_FLAGS_MINSIZEREL:STRING=
CMAKE_SHARED_LINKER_FLAGS_RELEASE:STRING=
CMAKE_SHARED_LINKER_FLAGS_RELWITHDEBINFO:STRING=
CMAKE_SKIP_INSTALL_RPATH:BOOL=NO
CMAKE_SKIP_RPATH:BOOL=NO
CMAKE_STATIC_LINKER_FLAGS:STRING=
CMAKE_STATIC_LINKER_FLAGS_DEBUG:STRING=
CMAKE_STATIC_LINKER_FLAGS_MINSIZEREL:STRING=
CMAKE_STATIC_LINKER_FLAGS_RELEASE:STRING=
CMAKE_STATIC_LINKER_FLAGS_RELWITHDEBINFO:STRING=
CMAKE_STRIP:FILEPATH=/usr/bin/strip
CMAKE_VERBOSE_MAKEFILE:BOOL=FALSE
CPACK_SOURCE_RPM:BOOL=OFF
CPACK_SOURCE_TBZ2:BOOL=ON
CPACK_SOURCE_TGZ:BOOL=ON
CPACK_SOURCE_TXZ:BOOL=ON
CPACK_SOURCE_TZ:BOOL=ON
CPACK_SOURCE_ZIP:BOOL=OFF
DVIPDF_CONVERTER:FILEPATH=/usr/bin/dvipdf
DVIPS_CONVERTER:FILEPATH=DVIPS_CONVERTER-NOTFOUND
GIT_EXECUTABLE:FILEPATH=/usr/bin/git
HTLATEX_COMPILER:FILEPATH=HTLATEX_COMPILER-NOTFOUND
LATEX2HTML_CONVERTER:FILEPATH=LATEX2HTML_CONVERTER-NOTFOUND
LATEX_COMPILER:FILEPATH=LATEX_COMPILER-NOTFOUND
LIBWEBSOCKETS_INCLUDE_DIR:PATH=/usr/local/include
LIBWEBSOCKETS_LIBRARIES:FILEPATH=/usr/local/lib/libwebsockets.so
LUALATEX_COMPILER:FILEPATH=LUALATEX_COMPILER-NOTFOUND
MAKEINDEX_COMPILER:FILEPATH=MAKEINDEX_COMPILER-NOTFOUND
OPENSSL_CRYPTO_LIBRARY:FILEPATH=/usr/local/ssl/lib/libcrypto.so
OPENSSL_INCLUDE_DIR:PATH=/usr/local/ssl/include
OPENSSL_SSL_LIBRARY:FILEPATH=/usr/local/ssl/lib/libssl.so
PDFLATEX_COMPILER:FILEPATH=PDFLATEX_COMPILER-NOTFOUND
PDFTOPS_CONVERTER:FILEPATH=/usr/bin/pdftops
PKG_CONFIG_EXECUTABLE:FILEPATH=/usr/bin/pkg-config
PS2PDF_CONVERTER:FILEPATH=/usr/bin/ps2pdf14
PYTHON_EXECUTABLE:FILEPATH=/usr/bin/python
SPHINX_EXECUTABLE:FILEPATH=/usr/bin/sphinx-build
UA_AMALGAMATION_ARCHITECTURES:STRING=
UA_ARCHITECTURE:STRING=posix
UA_ARCH_ADD_FLAGS:STRING=
UA_ARCH_EXTRA_INCLUDES:STRING=
UA_ARCH_LINKER_FLAGS:STRING=
UA_ARCH_REMOVE_FLAGS:STRING=
UA_BUILD_EXAMPLES:BOOL=OFF
UA_BUILD_FUZZING:BOOL=OFF
UA_BUILD_FUZZING_CORPUS:BOOL=OFF
UA_BUILD_OSS_FUZZ:BOOL=OFF
UA_BUILD_TOOLS:BOOL=OFF
UA_BUILD_UNIT_TESTS:BOOL=OFF
UA_DEBUG:BOOL=OFF
UA_DEBUG_DUMP_PKGS:BOOL=OFF
UA_ENABLE_AMALGAMATION:BOOL=ON
UA_ENABLE_CLANG_COV:BOOL=OFF
UA_ENABLE_COVERAGE:BOOL=OFF
UA_ENABLE_DA:BOOL=ON
UA_ENABLE_DETERMINISTIC_RNG:BOOL=OFF
UA_ENABLE_DISCOVERY:BOOL=ON
UA_ENABLE_DISCOVERY_MULTICAST:BOOL=OFF
UA_ENABLE_DISCOVERY_SEMAPHORE:BOOL=OFF
UA_ENABLE_ENCRYPTION:BOOL=ON
UA_ENABLE_ENCRYPTION_MBEDTLS:BOOL=OFF
UA_ENABLE_ENCRYPTION_OPENSSL:BOOL=ON
UA_ENABLE_EXPERIMENTAL_HISTORIZING:BOOL=OFF
UA_ENABLE_HARDENING:BOOL=ON
UA_ENABLE_HISTORIZING:BOOL=ON
UA_ENABLE_IMMUTABLE_NODES:BOOL=OFF
UA_ENABLE_JSON_ENCODING:BOOL=ON
UA_ENABLE_MALLOC_SINGLETON:BOOL=OFF
UA_ENABLE_METHODCALLS:BOOL=ON
UA_ENABLE_MICRO_EMB_DEV_PROFILE:BOOL=OFF
UA_ENABLE_NODEMANAGEMENT:BOOL=ON
UA_ENABLE_NODESET_COMPILER_DESCRIPTIONS:BOOL=ON
UA_ENABLE_PARSING:BOOL=ON
UA_ENABLE_PUBSUB:BOOL=ON
UA_ENABLE_PUBSUB_DELTAFRAMES:BOOL=ON
UA_ENABLE_PUBSUB_ETH_UADP:BOOL=ON
UA_ENABLE_PUBSUB_ETH_UADP_ETF:BOOL=OFF
UA_ENABLE_PUBSUB_ETH_UADP_XDP:BOOL=OFF
UA_ENABLE_PUBSUB_FILE_CONFIG:BOOL=OFF
UA_ENABLE_PUBSUB_INFORMATIONMODEL:BOOL=ON
UA_ENABLE_PUBSUB_INFORMATIONMODEL_METHODS:BOOL=ON
UA_ENABLE_PUBSUB_MONITORING:BOOL=ON
UA_ENABLE_PUBSUB_MQTT:BOOL=ON
UA_ENABLE_QUERY:BOOL=OFF
UA_ENABLE_STATIC_ANALYZER:STRING=OFF
UA_ENABLE_STATUSCODE_DESCRIPTIONS:BOOL=ON
UA_ENABLE_SUBSCRIPTIONS:BOOL=ON
UA_ENABLE_SUBSCRIPTIONS_ALARMS_CONDITIONS:BOOL=OFF
UA_ENABLE_SUBSCRIPTIONS_EVENTS:BOOL=OFF
UA_ENABLE_TYPEDESCRIPTION:BOOL=ON
UA_ENABLE_UNIT_TESTS_MEMCHECK:BOOL=OFF
UA_ENABLE_UNIT_TEST_FAILURE_HOOKS:BOOL=OFF
UA_ENABLE_VALGRIND_INTERACTIVE:BOOL=ON
**UA_ENABLE_WEBSOCKET_SERVER:BOOL=ON**
UA_FILE_NS0:BOOL=OFF
UA_FILE_NS0_BLACKLIST:BOOL=OFF
UA_FORCE_32BIT:BOOL=OFF
UA_FORCE_CPP:BOOL=OFF
UA_LOGLEVEL:STRING=300
UA_MSVC_FORCE_STATIC_CRT:BOOL=ON
UA_MULTITHREADING:STRING=0
UA_NAMESPACE_ZERO:STRING=REDUCED
UA_PACK_DEBIAN:BOOL=OFF
UA_VALGRIND_INTERACTIVE_INTERVAL:STRING=1000
XELATEX_COMPILER:FILEPATH=XELATEX_COMPILER-NOTFOUND
XINDY_COMPILER:FILEPATH=XINDY_COMPILER-NOTFOUND

```bash
cmake -DUA_NAMESPACE_ZERO=<YOUR_OPTION> <ANY_OTHER_OPTIONS> ..
```

## Checklist

Please provide the following information:

 - [X ] open62541 Version (release number or git tag): v1.2.1
 - [ ] Other OPC UA SDKs used (client or server): 
 - [ X] Operating system: raspbian
 - [ ] Logs (with `UA_LOGLEVEL` set as low as necessary) attached 
 - [ ] Wireshark network dump attached
 - [ ] Self-contained code example attached
 - [ X] Critical issue

**Comment by jackybek** [comment]
I forgot to mention that I successfully compiled Libwebsockets using the following:
git clone https://libwebsockets.org/repo/libwebsockets
cd libwebsockets; 
mkdir build;cd build
cmake ..
sudo make
sudo make install
sudo ldconfig
pkg-config --modversion libwebsockets

4.2.99-v4.2.0-5-g1a6c2a63

**Comment by jackybek** [comment]
If I temporarily comment out the statement and recompile as follows: 
//info.ws_ping_pong_interval = 10;

i get another error : compiler does not under UA_PubSubTransportLayerWS.  What should i use for websockets transport?

===================
SV_PubSub.c: In function ‘CreateServerPubSub’:
SV_PubSub.c:3607:38: warning: implicit declaration of function ‘UA_PubSubTransportLayerWS’; did you mean ‘UA_PubSubTransportLayerMQTT’? [-Wimplicit-function-declaration]
   config->pubsubTransportLayers[3] = UA_PubSubTransportLayerWS();
                                      ^~~~~~~~~~~~~~~~~~~~~~~~~
                                      UA_PubSubTransportLayerMQTT
SV_PubSub.c:3607:36: error: incompatible types when assigning to type ‘UA_PubSubTransportLayer’ {aka ‘struct <anonymous>’} from type ‘int’
   config->pubsubTransportLayers[3] = UA_PubSubTransportLayerWS();

**Comment by jackybek** [comment]
Surprise surprise!!!
Once i comment out the following:
/*
        #ifdef UA_ENABLE_WEBSOCKET_SERVER
          #ifdef UA_ENABLE_JSON_ENCODING
          {
                //transportProfile = UA_STRING(TRANSPORT_PROFILE_URI_WSSJSON);
                //networkAddressUrl.url = UA_STRING("opc.wss://192.168.1.11:7681/");    // points to a websocket proxy server;
                //config->pubsubTransportLayers[3] = UA_PubSubTransportLayerWS();
                //config->pubsubTransportLayersSize++;
          }
          #else
          {
                 //transportProfile = UA_STRING(TRANSPORT_PROFILE_URI_WSSBIN);
                 //networkAddressUrl.url = UA_STRING("opc.wss://192.168.1.11:7681/");      // points to a websocket proxy server;
                 //config->pubsubTransportLayers[3] = UA_PubSubTransportLayerWS();
                 //config->pubsubTransportLayersSize++;
          }
          #endif
        #endif
*/
and only has this in my program:

UA_ServerConfig_addNetworkLayerWS(UA_Server_getConfig(uaServer1), 7681, 0, 0, NULL, NULL) 

The websocket is successfully initialised on my OPCUA Server and I can now use a websocket client to access ws://192.168.1.109:7681

But I read somewhere that the websocket on OPCUA server should be access via opc.ws://192.168.1.109:7681 and not ws://192.168.1.109:7681.  Please advise.

Many thanks.

How come i am able to connect to OPCUA Server using ws://192.168.1.109:7681 ?

**Comment by jackybek** [comment]
Resolved in v1.3
